nominations/models.py

```diff
@@ -410,6 +546,64 @@ def save(self, *args, **kwargs):
         if self.vol_notes and not self.staff_notes:
             self.staff_notes = self.vol_notes
 
+    def create_related_objects(self):
+        """Create related nomination and questionnaire for application."""
+        if not self.nomination:
+            self.nomination = Nomination.objects.create()
+
+        if not self.questionnaire:
+            self.questionnaire = Questionnaire.objects.create()
+
+    def generate_application_status(self):
+        """Returns a generated status based on completion of various items.
+
+        nomination is filled out by the group with basic information about
+        the group and what it will do to help the candidate.
+
+        quesionnaire is filled out by the candidate with basic information and
+        in-depth policy positions.
+        """
+
+        if self.status in self.EDITABLE_STATUSES:
+            if self.nomination.status == 'incomplete':
+                if self.questionnaire.status == 'complete':
+                    status = 'needs-group-form'
+                else:
+                    status = 'needs-group-form-and-questionnaire'
+            else:
+                # nomination complete
+                if self.questionnaire.status == 'incomplete':
```

Comment:
@cjmabry i think there is a bug here - what about if questionnaire status is `sent`? This should be `if self.questionnaire.status != 'complete':`?